src/services/AnalysisService.ts

```diff
@@ -366,6 +368,86 @@ class AnalysisService {
     });
     if (status === ScanStatus.Incomplete || status === ScanStatus.Error) soosLogger.error(message);
   }
+
+  async findAnalysisFiles(
+    path: string,
+    pattern: string,
+    filesToExclude: string[] | null = null,
+    directoriesToExclude: string[] | null = null,
+    scanType: ScanType,
+  ): Promise<{ filePaths: string[]; hasMoreThanMaximumManifests: boolean }> {
```

Comment:
```suggestion
  async findAnalysisFiles(
    scanType: ScanType,
    path: string,
    pattern: string,
    filesToExclude: string[] | null = null,
    directoriesToExclude: string[] | null = null,
    maxFiles: number = 0,
  ): Promise<{ filePaths: string[]; hasMoreThanMaximumManifests: boolean }> {
```
scan type as first param, pass in the maxFiles - only check it if it is > 0